src/Deserialization/UnspentOutput.purs

```diff
@@ -80,7 +89,15 @@ convertOutput output = do
     dataHash =
       getDataHash maybeFfiHelper output <#>
         asOneOf >>> toBytes >>> T.DataHash
-  pure $ T.TransactionOutput { address, amount, dataHash }
+  datum <- getPlutusData maybeFfiHelper output # traverse convertPlutusData
```

Comment:
I would suggest to add an auxiliary function `convertOutputDatum` and utilize the CSL `.has_data_hash` and `.has_plutus_data` functions to explicitly show that we are working with alternative values